src/preferences/colorpalettesettings.cpp

```diff
@@ -0,0 +1,106 @@
+#include "preferences/colorpalettesettings.h"
+
+namespace {
+const mixxx::RgbColor kColorBlack(0x000000);
+const QString kColorPaletteConfigGroup = QStringLiteral("[Config]");
+const QString kColorPaletteGroup = QStringLiteral("[ColorPalette %1]");
+const QRegExp kColorPaletteGroupNameRegex("^\\[ColorPalette (.+)\\]$");
+const ConfigKey kHotcueColorPaletteConfigKey(kColorPaletteConfigGroup, QStringLiteral("HotcueColorPalette"));
+const ConfigKey kTrackColorPaletteConfigKey(kColorPaletteConfigGroup, QStringLiteral("TrackColorPalette"));
+
+int numberOfDecimalDigits(int number) {
+    int numDigits = 1;
+    while (number /= 10) {
+        numDigits++;
+    }
+    return numDigits;
+}
+
+} // anonymous namespace
+
+ColorPalette ColorPaletteSettings::getColorPalette(
+        const QString& name, const ColorPalette& defaultPalette) const {
+    QList<mixxx::RgbColor> colorList;
+
+    const QString group = kColorPaletteGroup.arg(name);
```

Comment:
It is better to use + here, it uses the string builder.
The same below.

Reply:
Ok. A shame that the StringBuilder syntax is so ugly :(

Reply:
const QString group = kColorPaletteGroup + name; 
already use the string builder. Do you consider this as ugly?